pkg/api/v1/pod/util.go

```diff
@@ -395,3 +399,47 @@ func UpdatePodCondition(status *v1.PodStatus, condition *v1.PodCondition) bool {
 	// Return true if one of the fields have changed.
 	return !isEqual
 }
+
+// UpdateTotalResources updates the TotalResourcesRequested field.
+// Returns true if pod TotalResourcesRequested has changed.
+func UpdateTotalResources(status *v1.PodStatus, pod *v1.Pod) bool {
+	requests := v1.ResourceList{}
+	// TODO: de-dupe this calculation
```

Comment:
Will update this PR or https://github.com/kubernetes/kubernetes/pull/115367, depending on which one goes in first.